vignettes/Treatment.Rmd

```diff
@@ -15,50 +15,188 @@ knitr::opts_chunk$set(
+
+Each drug parameter set is a vector of length four, that represent the drug efficacy, the infectiousness following treatment relative to an untreated infection, and parameters that determine the protection against reinfection ($P$): shape ($w$) and scale ($\lambda$) using an exponential decrease as follows:
+
+$$P = e^{{(-t/\lambda)}^w}$$
+
+The $P$ for each drug following treatment through time is plotted below.
+
+```{r, echo = FALSE, fig.align = 'center', fig.height = 5, fig.width = 7}
+cols  <- c("#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
+
+calc_P <- function(w, lambda, t){
+  P <- exp(-(t/lambda)^w)
+}
+
+t <- 1:70
+
+P_matrix <- lapply(list(AL_params, DHA_PQP_params, SP_AQ_params), function(x){
+  calc_P(w = x[3], lambda = x[4], t = t)
+})
+
+plot(x = t, y = P_matrix[[1]], type = "l", col = cols[1],
+     xlab = "Days", ylab = expression(paste("Protection against reinfection (",italic(P),")")))
+grid()
+
+# 
+invisible(sapply(2:3, function(x){
+  points(x = t, y = P_matrix[[x]], type = "l", col = cols[x])
+}))
+
+# Add legend
+legend("topright", legend = c("AL","DHA-PQP","SP-AQ"), col = cols, lty = 1)
+
+```
+
+For more details, please see:
+
+Okell, L., Cairns, M., Griffin, J. *et al.* Contrasting benefits of different artemisinin combination therapies as first-line malaria treatments using model-based cost-effectiveness analysis. *Nat Commun* **5**, 5606 (2014). <https://doi.org/10.1038/ncomms6606>.
+
+## Functions
+
+Drug parameters can be incorporated into a complete parameter set:
 
 ```{r}
+Intro_parms <- set_drugs(parameters = get_parameters(), drugs = AL_params)
+```
+
+A treatment regimen for each drug can then be described using the `set_clinical_treatment()` function which takes the drug index, a vector of timesteps at which a change in coverage occurs (where the initial coverage is 0 until the first timestep specified) and a vector of coverages for the drug that correspond with the timestep changes, as well as the complete parameter set.
+
+In this example, the AL coverage would be at 0% from the start until day 9, 20% during days 10-14, 50% during days 15-19 and 100% from day 20 until the end of the simulation.
+
+```{r}
+Intro_params <- set_clinical_treatment(parameters = Intro_parms,
+                                       drug = 1,
+                                       timesteps = c(10,15,20),
+                                       coverages = c(0.2,0.5,1))
+```
+
+Multiple drugs can be modelled simultaneously, with treatment coverage that can be specified for each drug. This function must be used for each drug included (e.g., if there are two drugs, `set_clinical_treatment()` must be called twice).
+
+# Example
```

Comment:
This takes a little time to run.
I wonder if a single run with one drug type, starting with 0% treatment coverage and then switching to high (90%?) treatment coverage halfway through would show the impact and be a bit more efficient?

Reply:
I've given a simpler example, but still included two drug regimens (I think that's important to show in an example).